--- gpm-1.20.7/src/daemon/startup.c.0004~	2017-05-31 18:34:29.528136326 +0200
+++ gpm-1.20.7/src/daemon/startup.c	2017-05-31 18:35:45.809317663 +0200
@@ -137,6 +137,11 @@ void startup(int argc, char **argv)
 
    //return mouse_table[1].fd; /* the second is handled in the main() */
 
+   // close extra fds
+   close(0);
+   close(1);
+   close(2);
+
    /****************** OLD CODE from gpn.c  END ***********************/
 
    init_mice(option.micelist);                  /* reads option.micelist */
